server/src/main/java/org/opensearch/search/query/ConcurrentQueryPhaseSearcher.java

```diff
@@ -104,8 +105,8 @@ private static boolean searchWithCollectorManager(
     }
 
     @Override
-    public AggregationProcessor newAggregationProcessor() {
-        return new ConcurrentAggregationProcessor();
+    public AggregationProcessor aggregationProcessor(SearchContext searchContext) {
+        return aggregationProcessor;
```

Comment:
I think we can remove this altogether and let it be handled by `QueryPhaseSearcherWrapper` but will defer until after #7970 is merged

Reply:
Yes, please, I will try to get it merged first